.github/workflows/check-submodule-pointers.yml

```diff
@@ -0,0 +1,25 @@
+name: Check submodule pointers
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-submodules:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Compare submodule pointers to lgtm.com branch
+        env: 
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+            if [ $(git rev-parse @:./ql) != $(gh api repos/github/codeql/git/ref/heads/lgtm.com --jq '.object.sha') ] || \
+               [ $(git rev-parse @:./codeql-go) != $(gh api repos/github/codeql-go/git/ref/heads/lgtm.com --jq '.object.sha') ]
+            then
+              echo "Submodules are out-of-date. Please update submodule pointers to the latest lgtm.com branch."
```

Comment:
Actions trick: `echo "::error error message here"` will highlight the error message visibly in the UI.
https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message